src/sentry/lang/javascript/processing.py

```diff
@@ -0,0 +1,146 @@
+from sentry.utils.safe import get_path
+
+logger = logging.getLogger(__name__)
+
+
+def _merge_frame_context(new_frame, symbolicated):
+    new_frame = dict(new_frame)
+    symbolicated = dict(symbolicated)
+
+    if symbolicated.get("pre_context"):
+        new_frame["pre_context"] = symbolicated["pre_context"]
+    if symbolicated.get("context_line"):
+        new_frame["context_line"] = symbolicated["context_line"]
+    if symbolicated.get("post_context"):
+        new_frame["post_context"] = symbolicated["post_context"]
+
+    return new_frame
+
+
+def _merge_frame(new_frame, symbolicated):
+    new_frame = dict(new_frame)
+    symbolicated = dict(symbolicated)
+
+    if symbolicated.get("function"):
+        new_frame["function"] = symbolicated["function"]
+    if symbolicated.get("abs_path"):
+        new_frame["abs_path"] = symbolicated["abs_path"]
+    if symbolicated.get("filename"):
+        new_frame["filename"] = symbolicated["filename"]
+    if symbolicated.get("lineno"):
+        new_frame["lineno"] = symbolicated["lineno"]
+    if symbolicated.get("colno"):
+        new_frame["colno"] = symbolicated["colno"]
+    if symbolicated.get("pre_context"):
+        new_frame["pre_context"] = symbolicated["pre_context"]
+    if symbolicated.get("context_line"):
+        new_frame["context_line"] = symbolicated["context_line"]
+    if symbolicated.get("post_context"):
+        new_frame["post_context"] = symbolicated["post_context"]
+    if symbolicated.get("status"):
+        frame_meta = new_frame.setdefault("data", {})
+        frame_meta["symbolicator_status"] = symbolicated["status"]
+
+    return new_frame
+
+
+# TODO: Change this error handling to be JS-specific?
+def _handle_response_status(event_data, response_json):
+    if not response_json:
+        error = SymbolicationFailed(type=EventError.NATIVE_INTERNAL_FAILURE)
+    elif response_json["status"] == "completed":
+        return True
+    elif response_json["status"] == "failed":
+        error = SymbolicationFailed(
+            message=response_json.get("message") or None, type=EventError.NATIVE_SYMBOLICATOR_FAILED
+        )
+    else:
+        logger.error("Unexpected symbolicator status: %s", response_json["status"])
+        error = SymbolicationFailed(type=EventError.NATIVE_INTERNAL_FAILURE)
+
+    write_error(error, event_data)
+
+
+def get_frames_for_symbolication(frames, data):
+    return [dict(frame) for frame in reversed(frames)]
+
+
+def is_sourcemap_image(image):
+    return (
+        bool(image)
+        and image.get("type") == "sourcemap"
+        and image.get("debug_id") is not None
+        and image.get("code_file") is not None
+    )
+
+
+def sourcemap_images_from_data(data):
+    return get_path(data, "debug_meta", "images", default=(), filter=is_sourcemap_image)
+
+
+def process_payload(data):
+    # We cannot symbolicate JS stacktraces without a release.
+    if data.get("release") is None:
+        return
```

Comment:
We will be able to with debug-ids :-)
But yes, this is pretty much blocked by being able to query artifact bundles by debug-id instead of release.